### PR TITLE
CHAT-1066: Make the chat into a drawer instead of the curent menu

### PR DESCRIPTION
**Pull request by hassenelkamel-exo**
To put the chat into drawer I add a new component `exoChatDrawer.vue` and I use the existent component `ExoChatContactList.vue` and `ExoChatMessageList.vue`.

### CODE REVIEW

File: application/src/main/webapp/vue-app/components/modal/ExoChatDrawer.vue [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<template>
+  <div id="drawerId" class="miniChatDrawer" @keydown.esc="closeChatDrawer()">
+    <a class="uiIconStatus uiNotifChatIcon" @click="openDrawer">
+    </a>
+    <div :class="showChatDrawer ? 'open' : '' " class="drawer">
+      <div class="header">
+        <span class="chatTitle">Chat</span>
+        <a class="closebtn" href="javascript:void(0)" @click="closeChatDrawer()">×</a>
+        <a v-exo-tooltip="$t('exoplatform.chat.open.chat')" data-placement="bottom" class="uiIconChatPopOut" href="/portal/intranet/chat" target="_chat"></a>
```
Comment (plamarque) [inline application/src/main/webapp/vue-app/components/modal/ExoChatDrawer.vue]: intranet can't bve hardcoded. in V6 the default portal will be /dw, not /intranet

Comment (hassenelkamel-exo) [reply application/src/main/webapp/vue-app/components/modal/ExoChatDrawer.vue]: done

File: application/src/main/webapp/vue-app/components/initComponents.js [hunk 2]
```diff
@@ -40,7 +40,8 @@ const components = {
   'exo-chat-room-notification-modal': ExoChatRoomNotificationModal,
   'exo-chat-notif-list': ExoMiniChatNotifList,
   'exo-chat-room': ExoMiniChatRoom,
-  'exo-chat-notif-detail': ExoMiniChatNotifDetail
+  'exo-chat-notif-detail': ExoMiniChatNotifDetail,
```
Comment (ahamdi) [inline application/src/main/webapp/vue-app/components/initComponents.js:43]: Shouldn't we cleanup old components related to MiniChat ?

File: application/src/main/webapp/vue-app/components/modal/ExoChatDrawer.vue [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+  name: 'ExoChatDrawer',
+  data () {
+    return {
+      showStatusUser : false,
+      contactList: [],
+      showSearch:false,
+      loadingContacts: true,
+      selectedContact: {},
+      userSettings: {
+        username: typeof eXo !== 'undefined' ? eXo.env.portal.userName : 'root'
```
Comment (ahamdi) [inline application/src/main/webapp/vue-app/components/modal/ExoChatDrawer.vue:54]: I think this is an old error, if username is undefined we can return empty string, but not root

File: application/src/main/webapp/vue-app/components/ExoChatContactList.vue [hunk 4]
```diff
@@ -204,6 +214,9 @@ export default {
       if (this.searchTerm && this.searchTerm.trim().length) {
         sortedContacts = sortedContacts.filter(contact => this.normalizeText(contact.fullName.toLowerCase()).indexOf(this.normalizeText(this.searchTerm.toLowerCase())) >= 0);
       }
+      if (this.searchWord && this.searchWord.trim().length) {
```
[the remark this answers is not in the document]
Comment (hassenelkamel-exo) [reply application/src/main/webapp/vue-app/components/ExoChatContactList.vue:217]: For the chat application before modification, the search is done by a filter ie by room, user ... for the drawer the search does not use a filter. so I add a new "searchTerm" to ensure the search in 2 ways.

File: application/src/main/webapp/vue-app/components/ExoChatApp.vue [hunk 5]
```diff
@@ -291,12 +292,15 @@ export default {
       if (tiptip) {
         tiptip.style.display = 'none';
       }
+    },
+    setStyle(){
+      if (location.pathname==='/portal/intranet/chat' || location.pathname==='/portal/dw/chat'){
```
[the remark this answers is not in the document]
Comment (hassenelkamel-exo) [reply application/src/main/webapp/vue-app/components/ExoChatApp.vue:297]: ok done!

File: application/src/main/webapp/vue-app/components/initComponents.js (hunk 2 above)
Comment (hassenelkamel-exo) [reply application/src/main/webapp/vue-app/components/initComponents.js:43]: yes indeed

File: application/src/main/webapp/vue-app/components/modal/ExoChatDrawer.vue (hunk 3 above)
Comment (hassenelkamel-exo) [reply application/src/main/webapp/vue-app/components/modal/ExoChatDrawer.vue]: yes done

File: application/src/main/webapp/vue-app/components/ExoChatContactList.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (ahamdi) [reply application/src/main/webapp/vue-app/components/ExoChatContactList.vue:217]: Please add a comment to explain the purpose of searchTerm and and searchWord and when each one is used.

File: application/src/main/webapp/vue-app/components/ExoChatContact.vue [hunk 6]
```diff
@@ -6,7 +6,7 @@
     </div>
     <div class="contactDetail">
       <div :class="isActive" class="contactLabel">
-        <span v-html="escapedName" />
+        <span class="userNameDrawer" v-html="escapedName" />
```
Comment (ahamdi) [inline application/src/main/webapp/vue-app/components/ExoChatContact.vue:9]: Please check if the class userNameDrawer is added to less files.

File: application/src/main/webapp/vue-app/components/ExoChatApp.vue [hunk 7]
```diff
@@ -291,12 +292,15 @@ export default {
       if (tiptip) {
         tiptip.style.display = 'none';
       }
+    },
+    setStyle(){
```
Comment (ahamdi) [inline application/src/main/webapp/vue-app/components/ExoChatApp.vue:296]:
```suggestion
    showHidePlatformAdminToolbar(){
```

File: application/src/main/webapp/vue-app/components/ExoChatContact.vue (hunk 6 above)
Comment (ahamdi) [reply application/src/main/webapp/vue-app/components/ExoChatContact.vue:9]: Is this CSS class added somewhere ?

Comment (hassenelkamel-exo) [reply application/src/main/webapp/vue-app/components/ExoChatContact.vue:9]: no, i remove it .

File: application/src/main/webapp/css/components/chatContact.less [hunk 8]
```diff
@@ -46,22 +46,22 @@
 
 .user-invisible, .user-offline {
   .user-status, .uiIconStatus, .uiIconStatus:hover, > a .uiIconStatus, &:hover .user-status, &:hover .uiIconStatus {
-    color: #999999;
+    color: #999999!important;
```
Comment (boubaker) [inline application/src/main/webapp/css/components/chatContact.less:49]: Sorry, I didn't communicated yet about the best practice for defining Colors and how to use it.
Same thing for common Vue components.
anyway, it's better IMO to reuse variables for colors coming from platform-ui, see perk-store, wallet, social ...
This color should be reused from variables.less coming from platform-ui. Thanks

Comment (ahamdi) [reply application/src/main/webapp/css/components/chatContact.less:49]: @hassenelkamel-exo please take care of this before merging

Comment (ahamdi) [reply application/src/main/webapp/css/components/chatContact.less:49]: @boubaker We will merge and start testing the current PR , in the meantime, @hassenelkamel-exo proposed a PR to consider your feedback, h ewill sedn it for review. Thanks

Comment (hassenelkamel-exo) [reply application/src/main/webapp/css/components/chatContact.less:49]: ok, thanks

**Pull request merged**